Wrappers/Python/test/test_Gradient.py

```diff
@@ -17,555 +17,382 @@
+        N, M, K = 10, 11, 12
+        channels = 13
 
+        self.voxel_size_x, self.voxel_size_y, self.voxel_size_z = 0.1, 0.3, 0.6
 
-        #reordered
-        data.reorder(['vertical','horizontal_x','channel','horizontal_y'])
-        grad_py = GradientOperator(data.geometry, bnd_cond='Neumann', correlation='SpaceChannels', backend='numpy')
-        gold_direct = grad_py.direct(data)
-        gold_adjoint = grad_py.adjoint(gold_direct)
+        self.ig_2D = ImageGeometry(voxel_num_x = M, voxel_num_y = N) 
+        self.ig_2D_chan = ImageGeometry(voxel_num_x = M, voxel_num_y = N, channels = channels) 
+        self.ig_3D = ImageGeometry(voxel_num_x = M, voxel_num_y = N, voxel_num_z= K) 
+        self.ig_3D_chan = ImageGeometry(voxel_num_x = M, voxel_num_y = N, channels = channels, voxel_num_z= K) 
 
-        grad_c = GradientOperator(data.geometry, bnd_cond='Neumann', correlation='SpaceChannels', backend='c')
-        out_direct = grad_c.direct(data)
-        out_adjoint = grad_c.adjoint(out_direct)
+        self.ig_2D_voxel = ImageGeometry(voxel_num_x = M, voxel_num_y = N, voxel_size_x=self.voxel_size_x, voxel_size_y=self.voxel_size_y) 
+        self.ig_2D_chan_voxel = ImageGeometry(voxel_num_x = M, voxel_num_y = N, channels = channels, voxel_size_x=self.voxel_size_x, voxel_size_y=self.voxel_size_y) 
+        self.ig_3D_voxel = ImageGeometry(voxel_num_x = M, voxel_num_y = N, voxel_num_z= K, voxel_size_x=self.voxel_size_x, voxel_size_y=self.voxel_size_y, voxel_size_z=self.voxel_size_z) 
+        self.ig_3D_chan_voxel = ImageGeometry(voxel_num_x = M, voxel_num_y = N, channels = channels, voxel_num_z= K, voxel_size_y=self.voxel_size_y, voxel_size_x=self.voxel_size_x, voxel_size_z=self.voxel_size_z)         
 
-        #print("GradientOperator, 4D, bnd_cond='Neumann', direct")
-        numpy.testing.assert_array_equal(out_direct.get_item(0).as_array(), gold_direct.get_item(0).as_array())
-        numpy.testing.assert_array_equal(out_direct.get_item(1).as_array(), gold_direct.get_item(1).as_array())
-        numpy.testing.assert_array_equal(out_direct.get_item(2).as_array(), gold_direct.get_item(2).as_array())
-        numpy.testing.assert_array_equal(out_direct.get_item(3).as_array(), gold_direct.get_item(3).as_array())
+        self.list_geometries = [self.ig_2D, self.ig_2D_chan, self.ig_3D, self.ig_3D_chan,
+                                self.ig_2D_voxel, self.ig_2D_chan_voxel, self.ig_3D_voxel, self.ig_3D_chan_voxel]
+        self.bconditions = ['Neumann', 'Periodic']                                
+        self.backend = ['numpy','c']
+        self.correlation = ['Space','SpaceChannels']
+        self.method = ['forward', 'backward', 'centered']
 
-        #print("GradientOperator, 4D, bnd_cond='Neumann', adjoint")
-        numpy.testing.assert_array_equal(out_adjoint.as_array(), gold_adjoint.as_array())
+        super(TestGradientOperator, self).__init__(*args, **kwargs)
```

Comment:
the `super` call is not required once you have used `setUp` instead of `__init__`

Reply:
fixed